build.gradle

```diff
@@ -124,24 +122,34 @@ gradle.projectsEvaluated {
     }
 }
 
+sonatypeCentralUpload {
+    // This is your Sonatype generated username
+    username = System.getenv("SONATYPE_USERNAME")
+    // This is your sonatype generated password
+    password = System.getenv("SONATYPE_PASSWORD")
 
-publishing {
-    repositories {
-        maven {
-            name = "OSSRH"
-            url = 'https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/'
-            credentials {
-                username = System.getenv("MAVEN_USERNAME")
-                password = System.getenv("MAVEN_PASSWORD")
-          }
-        }
-    }
+    // This is a list of files to upload. Ideally you would point to your jar file, source and javadoc jar (required by central)
+    archives = files(
+      "$buildDir/libs/${artifactId}-${version}.jar",
+      "$buildDir/libs/${artifactId}-${version}-sources.jar",
+      "$buildDir/libs/${artifactId}-${version}-javadoc.jar"
+    )
 
+    // This is the pom file to upload. This is required by central
+    pom = file("$buildDir/pom.xml")
+
+    // This is your PGP private key. This is required to sign your files
+    signingKey = System.getenv("SONATYPE_SIGNING_KEY")
+    // This is your PGP private key passphrase to decrypt your private key
+    signingKeyPassphrase = System.getenv("SIGNING_KEY_PASSPHRASE")
+}
+
+publishing {
     publications {
         maven(MavenPublication) {
-            groupId = 'com.styra.opa'
-            artifactId = 'springboot'
-            version = version
+            groupId = "${groupId}"
+            artifactId = "${artifactId}"
+            version = "${version}"
```

Comment:
Uhm, where are these set? If it was previously OK to do `version = version`, why do we need the `"${version}"` now? 💭